### PR TITLE
feat(Audible): new presence

### LINKED ISSUE #5479: Audible | audible.com
### Discussed in https://github.com/PreMiD/Presences/discussions/4433

<div type='discussions-op-text'>

<sup>Originally posted by **LagSwitchedVirginity** November 30, 2019</sup>
**What's the name of the service?**
Ex. https://www.audible.com/ | Audible

**What should the Presence display?**
Informations about the current audio book

**Why do you want support for this service?**
Audible is a audio book service, I and a good amount of friends listen to books, and when people ask me on DND "what are you doing?" instead of me having to pause the book, and respond, it would be nice for it to just be evident. :)

**Are you able to provide an image for the service? (512x512 minimum)**
https://www.logolynx.com/images/logolynx/96/96b51772804f1345e573061484f85693.png</div>

### PR DESCRIPTION
**Pull request by realtofuine**
## Description 
add a new presence for Audible (audible.com)

closes #5479 

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

image

</details>

### CODE REVIEW

File: websites/A/Audible/presence.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
+	};
+
+	if (document.location.pathname.includes("/account")) {
+		presenceData.details = "Viewing account";
+		delete presenceData.state;
```
Comment (ImRodry) [inline websites/A/Audible/presence.ts:13]: state wasn't assigned a value prior to this, this isn't needed

File: websites/A/Audible/presence.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		presenceData.startTimestamp = browsingTimestamp;
```
Comment (ImRodry) [inline websites/A/Audible/presence.ts:14]: just assign this at the object declaration

**Review by ImRodry — COMMENTED** [review]
Apply that suggestion to all the cases
Also are you sure you're supporting every page on the website?

**Review by ImRodry — CHANGES_REQUESTED** [review]
Oopsies

File: websites/A/Audible/presence.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		clientId: "991933219719086080",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
+	};
+
+	presenceData.startTimestamp = browsingTimestamp;
```
Comment (EncryptedDev) [inline websites/A/Audible/presence.ts:11]:
```suggestion
presence.on("UpdateData", async () => {
	const presenceData: PresenceData = {
		largeImageKey: "logo",
		startTimestamp: browsingTimestamp
	};
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/A/Audible/presence.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+			).textContent,
+			cover = document
+				.querySelector("img.bc-pub-block.bc-image-inset-border.js-only-element")
+				.getAttribute("src");
+		presenceData.details = `Viewing ${title} by ${
+			document.querySelector(
+				"li.bc-list-item.authorLabel > a.bc-link.bc-color-link"
+			).textContent
+		}`;
+		presenceData.largeImageKey = cover;
```
Comment (EncryptedDev) [inline websites/A/Audible/presence.ts:33]: Apply the same logic from here https://github.com/PreMiD/Presences/pull/6408/files#r917325055

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**